lib/seahorn/BvOpSem2MemRepr.cc

```diff
@@ -183,6 +265,24 @@ OpSemMemRepr::MemValTy OpSemMemArrayRepr::MemFill(PtrTy dPtr, char *sPtr,
   return MemValTy(res);
 }
 
+Expr OpSemMemHybridRepr::loadAlignedWordFromMem(PtrTy ptr, MemValTy mem) {
+  LOG("opsem-hybrid", INFO << "Load ptr " << *ptr.toExpr() << "\n");
+  LOG("opsem-hybrid", INFO << "From ptr " << *mem.toExpr() << "\n");
+
+  /** rewrite store into ITE **/
+  ArrayStoreRewriter rw(ptr.toExpr(), m_ctx.alu(),
```

Comment:
IIUC when we `load` we lazily rewrite the `store` instructions in the memory array rather than rewriting everytime we store... is this correct?

Reply:
yes. optimizing while doing so. next layer is to optimize store instructions to simplify load optimizations later on